service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -94,10 +95,31 @@ public void initializeSession(IoSession session, ConnectFuture future) {
             HttpConnectSession connectSession = (HttpConnectSession) session;
             connectSession.setVersion(acceptSession.getVersion());
             connectSession.setMethod(acceptSession.getMethod());
-            connectSession.setRequestURI(acceptSession.getRequestURI());
+            if (!validateRequestPath()) {
+                acceptSession.setStatus(CLIENT_NOT_FOUND);
+                acceptSession.close(false);
+                return;
+            }
+            String acceptPathInfo = acceptSession.getPathInfo().toString();
+            if (acceptPathInfo.indexOf("/") == 0) {
```

Comment:
Is this if condition needed at all ? If the accept session requstURI doesn't contain /, then it would be wrong at http transport level.